src/pretalx/orga/views/settings.py

```diff
@@ -33,7 +41,11 @@ def dispatch(self, request, *args, **kwargs):
         if self._action == 'create':
             if not request.user.is_anonymous and not request.user.is_superuser:
                 raise PermissionDenied()
-        return super().dispatch(request, *args, **kwargs)
+        ret = super().dispatch(request, *args, **kwargs)
+        if self._action != 'create':
```

Comment:
There is no open transaction here. If I am not mistaken, any changes caused by the `dispatch()` above will not be reverted when we raise an Exception below. So the users sees an error page, but the changes are saved nonetheless. A quick and dirty test seems to confirm that.

This check does not seem to depend on the result of `dispatch()`, so move the if above the call, just to be sure?

Reply:
Yup, that's right.